### DISCUSSION TITLE
Result["face_index"].AsInt32() is return '-1' in godot, while I'm using joltphysics

### DISCUSSION THREAD

**Discussion by chaosjoey** [jrouwe/JoltPhysics#1317] [Q&A]
I'm making a game in Godot, I recently imported the GodotBolt addon, and I've encountered a problem, where I used to use raycasts to get UV coordinates, but implementing bolts seems to have broken that. When I try to access the face index from a raycast result I always get '-1'

I've seen similar discussions, with solutions involving using or setting "MeshShapeSettings::mPerTriangleUserData" to true. But to be honest I'm really out of my depth and have no idea how to do that. Could really use some help. I don't know if this is something I should be setting in Godot's settings and preferences, or if this is something I should be changing or using in my scripts.
I'm using Godot, typically coding with C#. Though I know C++.

For reference here's the sources that are making me think I need to do something with "MeshShapeSettings::mPerTriangleUserData"
https://github.com/jrouwe/JoltPhysics/blob/master/Docs/ReleaseNotes.md
https://github.com/jrouwe/JoltPhysics/pull/1225/commits/ce1505382938d9f813ce098700761d22e8698f61
https://github.com/godot-jolt/godot-jolt/issues/702

Sorry if this is a very nooby question. I promise I've had years of classes in programming, this is the most confused I've been in a while

**Comment by mihe** [comment]
No new release of Godot Jolt has been cut since the support for `face_index` was added (in godot-jolt/godot-jolt#955), so the only way to gain access to it currently is to [build the extension from source](https://github.com/godot-jolt/godot-jolt/blob/master/docs/building.md) yourself. I am however in the process of making a new release as we speak, so there should hopefully be a 0.14.0-stable out later today.

Note that `face_index` will still be opt-in through the `physics/jolt_3d/queries/enable_ray_cast_face_index` project setting, as it comes with a fairly heavy memory cost, as talked about in godot-jolt/godot-jolt#702.

Also, please post any future Godot-related issues/discussions in the Godot Jolt repository instead.

**Reply by chaosjoey** [reply]
Oh thanks! Sorry for post in the wrong spot didn't know there were separate repositories. I will be sure to post any godot related stuff in the godot one in the future!
